precompiles/bank/bank.go

```diff
@@ -217,10 +219,43 @@ func (p Precompile) sendNative(ctx sdk.Context, method *abi.Method, args []inter
 		return nil, err
 	}
 
+	if hooks := tracers.GetCtxEthTracingHooks(ctx); hooks != nil && hooks.OnBalanceChange != nil && !wei.IsZero() {
```

Comment:
note that `GetWeiBalance` will only give you the sub-usei part of the balance change (e.g. if the transferred amount is 1000000000001wei, `GetWeiBalance` will return 1, and the rest are in `GetBalance(ctx, addr, ctx.GetBaseDenom()` (which will return 1 because 1000000000000wei is 1usei)

Reply:
Ok, so indeed that's far from what we want here :) I'm going to work on the aspect now to ensure I've got them all correct, thanks!